src/UBAFeeCalculator/UBAFeeCalculator.test.ts

```diff
@@ -56,46 +58,117 @@ describe("UBA Fee Calculations", () => {
     const [idx, [lowerBound, upperBound]] = getInterval(tuples, toBN("12000000"));
     expect(idx).to.eq(14);
     expect(lowerBound.toString()).to.eq("9000000");
-    expect(upperBound.toString()).to.eq(MAX_SAFE_JS_INT.toString());
+    expect(upperBound.toString()).to.eq(BigNumber.from(MAX_SAFE_JS_INT).mul(parseEther("1")).toString().toString());
   });
 
   it("should integrate the correct value: test #1", () => {
     const result = performLinearIntegration(tuples, 0, toBN(0), toBN(100_000));
-    expect(result.div(parseEther("1")).toString()).to.eq("-40000");
+    expect(result.toString()).to.eq("-40000");
   });
 
   it("should integrate the correct value: test #2", () => {
     const result = performLinearIntegration(tuples, 0, toBN(100_000), toBN(0));
-    expect(result.div(parseEther("1")).toString()).to.eq("40000");
+    expect(result.toString()).to.eq("40000");
   });
 
   it("should integrate the correct value: test #3", () => {
     const result = performLinearIntegration(tuples, 1, toBN(100_000), toBN(250_000));
-    expect(result.div(parseEther("1")).toString()).to.eq("-48750");
+    expect(result.toString()).to.eq("-48750");
   });
 
   it("should integrate the correct value: test #4", () => {
     const result = performLinearIntegration(tuples, 1, toBN(250_000), toBN(100_000));
-    expect(result.div(parseEther("1")).toString()).to.eq("48750");
+    expect(result.toString()).to.eq("48750");
   });
 
   it("should compute the correct deposit fee #1", () => {
     const result = getDepositBalancingFee(tuples, toBN(300_000), toBN(50_000));
-    expect(result.div(parseEther("1")).toString()).to.eq("-10250");
+    expect(result.toString()).to.eq("-10250");
   });
 
   it("should compute the correct deposit fee #2", () => {
     const result = getDepositBalancingFee(tuples, toBN(300_000), toBN(100_000));
-    expect(result.div(parseEther("1")).toString()).to.eq("-19000");
+    expect(result.toString()).to.eq("-19000");
   });
 
   it("should compute the correct refund fee #1", () => {
     const result = getRefundBalancingFee(tuples, toBN(350_000), toBN(50_000));
-    expect(result.div(parseEther("1")).toString()).to.eq("10250");
+    expect(result.toString()).to.eq("10250");
   });
 
   it("should compute the correct refund fee #2", () => {
     const result = getRefundBalancingFee(tuples, toBN(300_000), toBN(100_000));
-    expect(result.div(parseEther("1")).toString()).to.eq("25500");
+    expect(result.toString()).to.eq("25500");
+  });
+});
+
+describe("UBA Fee Calculations from Data", () => {
+  let gammaCutoffArray: FlowTupleParameters;
+  let omegaCutoffArray: FlowTupleParameters;
+
+  beforeEach(() => {
+    gammaCutoffArray = [
```

Comment:
Why set these arrays in the `beforeEach` and why not in a single `before()`?

Reply:
Good call - I've updated to a single `beforeAll` instead of the `beforeEach`.